src/libsystemd/sd-login/sd-login.c

```diff
@@ -389,10 +389,12 @@ _public_ int sd_session_get_uid(const char *session, uid_t *ret_uid) {
 }
 
 _public_ int sd_session_get_username(const char *session, char **ret_username) {
+        assert_return(ret_username, -EINVAL);
         return session_get_string(session, "USER", ret_username);
```

Comment:
session_get_string() already has the same assert...

if functions encapsulate other basically funcs 1:1 and do not actually contain any code except for augmenting some arguments then I don#t think we neeed to repeat the asserts. I'd just leave that to the inner, common function